### PR TITLE
[Improvement][Helm] Refactor configmap control in master

### LINKED ISSUE #15934: [Improvement][Helm] Refactor configmap control in master
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by pegasas**
## Purpose of the pull request

fix: https://github.com/apache/dolphinscheduler/issues/15934

## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

(or)

If your pull request contain incompatible change, you should also add it to `docs/docs/en/guide/upgrede/incompatible.md`

### CODE REVIEW

**Comment by ruanwenjun** [comment]
@Gallardot PTAL

File: deploy/kubernetes/dolphinscheduler/templates/statefulset-dolphinscheduler-master.yaml [hunk 1]
```diff
@@ -109,11 +109,9 @@ spec:
           volumeMounts:
             - mountPath: "/opt/dolphinscheduler/logs"
               name: {{ include "dolphinscheduler.fullname" . }}-master
-            {{- include "dolphinscheduler.sharedStorage.volumeMount" . | nindent 12 }}
             - name: config-volume
               mountPath: /opt/dolphinscheduler/conf/common.properties
               subPath: common.properties
-            {{- include "dolphinscheduler.etcd.ssl.volumeMount" . | nindent 12 }}
```
Comment (Gallardot) [inline deploy/kubernetes/dolphinscheduler/templates/statefulset-dolphinscheduler-master.yaml:116]: Why delete the ssl volume of etcd?

ref: #13924   
@qingwli PTAL.

File: deploy/kubernetes/dolphinscheduler/templates/statefulset-dolphinscheduler-master.yaml [hunk 2]
```diff
@@ -109,11 +109,9 @@ spec:
           volumeMounts:
             - mountPath: "/opt/dolphinscheduler/logs"
               name: {{ include "dolphinscheduler.fullname" . }}-master
-            {{- include "dolphinscheduler.sharedStorage.volumeMount" . | nindent 12 }}
```
Comment (Gallardot) [inline deploy/kubernetes/dolphinscheduler/templates/statefulset-dolphinscheduler-master.yaml:112]: That's fine with me. Why not remove this from `api-server` as well.

File: deploy/kubernetes/dolphinscheduler/templates/statefulset-dolphinscheduler-master.yaml (hunk 1 above)
Comment (pegasas) [reply deploy/kubernetes/dolphinscheduler/templates/statefulset-dolphinscheduler-master.yaml:116]: it seems this is also needed in `api`, 
so I kept this and adjust the order to make it clear to read.

File: deploy/kubernetes/dolphinscheduler/templates/statefulset-dolphinscheduler-master.yaml (hunk 2 above)
Comment (pegasas) [reply deploy/kubernetes/dolphinscheduler/templates/statefulset-dolphinscheduler-master.yaml:112]: it seems this config is needed,
I adjust the order and fix some config in `worker`

File: deploy/kubernetes/dolphinscheduler/templates/statefulset-dolphinscheduler-worker.yaml [hunk 3]
```diff
@@ -133,11 +133,6 @@ spec:
           {{- else }}
           emptyDir: {}
           {{- end }}
-        {{- else }}
-        - name: {{ include "dolphinscheduler.fullname" . }}-worker-data
-          emptyDir: {}
-        - name: {{ include "dolphinscheduler.fullname" . }}-worker-logs
-          emptyDir: {}
```
Comment (Radeity) [inline deploy/kubernetes/dolphinscheduler/templates/statefulset-dolphinscheduler-worker.yaml:140]: Seems like these lines shouldn't be removed?

File: deploy/kubernetes/dolphinscheduler/templates/statefulset-dolphinscheduler-master.yaml [hunk 4]
```diff
@@ -122,10 +122,10 @@ spec:
           {{- else }}
           emptyDir: {}
           {{- end }}
-        {{- include "dolphinscheduler.sharedStorage.volume" . | nindent 8 }}
         - name: config-volume
           configMap:
             name: {{ include "dolphinscheduler.fullname" . }}-configs
+        {{- include "dolphinscheduler.sharedStorage.volume" . | nindent 8 }}
```
Comment (qingwli) [inline deploy/kubernetes/dolphinscheduler/templates/statefulset-dolphinscheduler-master.yaml:128]: Just change from line115  to line 128?

File: deploy/kubernetes/dolphinscheduler/templates/statefulset-dolphinscheduler-worker.yaml (hunk 3 above)
Comment (qingwli) [reply deploy/kubernetes/dolphinscheduler/templates/statefulset-dolphinscheduler-worker.yaml:140]: > Seems like these lines shouldn't be removed?

+1

**Pull request closed without merging**